### ISSUE TITLE
🎙 talk.js - August 2019

### ISSUE THREAD

**Issue by ongmin** [SingaporeJS/talk.js#25]
### 📝 Details
- [x] Event: [talk.js](http://meetu.ps/e/H46BJ/10jMR/a)
- [x] Date: 14 Aug (Wed)
- [x] Hosts: @ongmin
- [x] Venue: Funding Societies, Keppel Tower #05-08, 10 Hoe Chiang Road, Singapore 089315.

### 📅 Agenda

Time   | Agenda
------ | :-----
7.00pm | Doors open - _mingle with guests and speakers over 🍕_
7.30pm | Start of event - _opening segment by host_
7.40pm | Start of scheduled talks
8.40pm | Open announcements - _open segment for anyone to make announcements_
9.00pm | End of event

### :speech_balloon: Talks

- [ ] My Awesome Talk by _You_

### :seedling: Propose a talk this month

Want to speak at this month's talk? Leave a comment below with the following details:

<details>
<summary>Click for details</summary>

> #### Title of your topic
> Insert catchy title here
>
> #### Topic description
> Tell us what you'd like to talk about
>
> #### Presenter(s)
> Include GitHub handles where possible. Add your Twitter handle if you wish
>
> #### Presenter's bio
> A short introduction of yourself
>
> #### Duration
> 5 minutes for presenting an introduction to the topic, then up to 20 minutes for discussion.
>
> #### Deck URL (if any)
> You can add this part in later

</details>

### :loudspeaker: Open announcements
Have something you'd like to tell the audience? We have an open announcement segment for you to announce job opportunities, promote a meet up, upcoming conferences, etc.

---

### :pencil: Organizers' checklist

<details>
<summary>Click for details</summary>

- [x] Check available dates on [EngineersSG](https://engineers.sg/)
- [x] Confirm date and venue availability
- [ ] Create event on [SingaporeJS Meetup Page](https://www.meetup.com/singapore-js/)
- [ ] Inform @engineersftw for video recordings
- [ ] Announce event on [SingaporeJS Gitter](https://gitter.im/SingaporeJS/home) and other social media platforms
- [ ] Prepare deck for event

\* _checked boxes indicate confirmed/done_

</details>

**Comment by linusnorton** [comment]
**Title of your topic**

Node.js architecture - guiding principles

**Topic description**

An examination of which programming principles from the object oriented and functional programming worlds apply to node.js, and how they guide the design and structure of an application.

**Presenter(s)**

Linus Norton - @linusnorton

**Presenter's bio**

Linus Norton has a background of over 10 years in the UK rail industry and now works for the Ministry of Justice in London as a full stack engineer. An equal part experience of algorithmic coding and domain driven design has led to a blend of functional and object oriented programming principles that can be at odds with each other.  

**Duration**

25-30 minutes for presenting plus discussion.

**Comment by pkamath2** [comment]
**Title of your topic:** 
Generative Art using Genetic Algorithms and Neural networks

**Topic description**
A talk based on creating a Generative Art system using Genetic Algorithms and Neural Networks. This idea to create abstract patterns is based out of Genetic or Evolution based Art systems like DNGA, SNGA or Picbreeder, but created in Javascript using Tensorflow.js, Node & D3. 

**Presenter(s)**
Purnima Kamath
(@purnimakamath) https://twitter.com/purnimakamath

**Presenter's bio**
Purnima is an engineer, programming Java based web and data engineering systems. She is also an amateur generative artist and painter in oils. She is a Director on the Women Who Code network and an evangelist for YOW! Conference in Singapore.

**Duration**
20-25 mins presentation 5 mins Q&A if time allows. 

**Deck URL (if any)**
TBC

**Comment by yishus** [comment]
Title of your topic

An exploration of CJSS - A CSS based web framework

Topic description

What started out as an extended joke, Scott Kellum created a framework where everything can be written in CSS. We will explore the codebase that he has open sourced, and a retrospective look into web frameworks.

Presenter(s)

Yishu - github @yishus twitter @yishusee

Presenter's bio

Yishu sometimes likes to fill up google forms, sometimes not.

Duration

20min

**Comment by mikhfilippov** [comment]
Title of your topic
What Babylonian Astronomers Could Learn About Patterns in Javascript Code?

Topic description
Software engineering collaboration is hard. Software engineers spend > 70% of their time learning about their own team’s source code. Enormous size, constant change and intricate dependencies of the source code - are the main reason causing that problem.

To help software engineers at Quod AI we translate source code into natural language to make it easier to search, navigate and understand. In order to do that we use neural network models, natural language processing algorithms and statistical models.

In this talk we will share some of the insights that we gained from analyzing more than 300 millions lines of code.

Presenter(s)
@mikhfilippov

Presenter's bio
I am an applied mathematician, who is building an AI platform that explains source code in plain English.

Duration
I guess 20 minutes presentation, then 10 minutes Q+A

**Comment by linusnorton** [comment]
That works for me, I can certainly condense my talk to 20 minutes if someone would like an extra 5.

**Comment by mikhfilippov** [comment]
Sounds fine for me. Thank you @ongmin

**Comment by yishus** [comment]
okay!

**Comment by pkamath2** [comment]
Hey @ongmin, Unfortunately I may need to travel out of Singapore during that week and will not be able to speak this time =(( Apologies for the short notice. 
Any chance I can move this talk to another month?

**Comment by ongmin** [comment]
@pkamath2, as chatted, sure no problem. :)

**Comment by linusnorton** [comment]
Do we have a venue?

**Comment by ongmin** [comment]
@linusnorton, oh yes! I updated it on meet-up.com, haven’t updated it here. It’s at Funding Societies. Will update this soon.

**Comment by ongmin** [comment]
Links for @linusnorton's talk:
https://docs.google.com/presentation/d/1cjQPT5er14fPQ6cIIy6fJVF6yRv8rFrgXOAuj_GzlOc/edit?usp=sharing
https://ljn.io/posts/node-architecture/
https://ljn.io/posts/encapsulating-mutable-state/

**Comment by ongmin** [comment]
Hi @yishus @mikhfilippov, if you would like to add your decks to this issue, please feel free to! I'll keep this thread open longer. :)
